ts/src/coindcx.ts

```diff
@@ -0,0 +1,1754 @@
+        //             "pnl": 0.0,
+        //             "initial_margin": 0.02418828,
+        //             "interest": 0.0667,
+        //             "interest_amount": 0.0,
+        //             "interest_amount_updated_at": 0,
+        //             "interest_free_hours": 1.0,
+        //             "leverage": 1.0,
+        //             "result": null,
+        //             "tds_amount": null,
+        //             "margin_tds_records": [],
+        //             "created_at": 1718540754877,
+        //             "updated_at": 1718540754877,
+        //             "orders": [
+        //                 {
+        //                     "id": 104430986,
+        //                     "order_type": "market_order",
+        //                     "status": "initial",
+        //                     "market": "ACHUSDT",
+        //                     "side": "buy",
+        //                     "avg_price": 0.0,
+        //                     "total_quantity": 1.0,
+        //                     "remaining_quantity": 1.0,
+        //                     "price_per_unit": 0.0,
+        //                     "timestamp": 1718540754920.9573,
+        //                     "maker_fee": 0.1,
+        //                     "taker_fee": 0.1,
+        //                     "fee": 0.1,
+        //                     "fee_amount": 0.0,
+        //                     "filled_quantity": 0.0,
+        //                     "bo_stage": "stage_entry",
+        //                     "cancelled_quantity": 0.0,
+        //                     "stop_price": 0.0
+        //                 }
+        //             ]
+        //         }
+        //     ]
+        //
+        const position = this.safeDict (response, 0, {});
+        const orders = this.safeList (position, 'orders', []);
+        const order = this.safeDict (orders, 0, {});
+        const parsedOrder = this.parseOrder (order, market);
+        const id = this.safeString (position, 'id'); // using id of the position as id of the order for user could fetch or cancel it
+        if (id !== undefined) {
+            parsedOrder['id'] = id;
+        }
+        parsedOrder['info'] = position;
+        return parsedOrder;
+    }
+
+    encodeSpotOrderType (type) {
+        const types = {
+            'market': 'market_order',
+            'limit': 'limit_order',
+            'market_order': 'market_order',
+            'limit_order': 'limit_order',
+        };
+        return this.safeString (types, type, undefined);
+    }
+
+    encodeMarginOrderType (type) {
+        const types = {
+            'market': 'market_order',
+            'limit': 'limit_order',
+            'market_order': 'market_order',
+            'limit_order': 'limit_order',
+            'stop_limit': 'stop_limit',
+            'take_profit': 'take_profit',
+            'take_profit_limit': 'take_profit',
+        };
+        return this.safeString (types, type, undefined);
```

Comment:
if you do `return this.safeString (types, type, type)` here you just need to list the different ones because you're defaulting to the input